CRM/Event/Cart/Form/Checkout/Payment.php

```diff
@@ -263,14 +263,14 @@ public function process_event_line_item(&$event_in_cart, $class = NULL) {
       $price_set = $price_sets[$price_set_id];
       $price_set_amount = array();
       CRM_Price_BAO_PriceSet::processAmount($price_set['fields'], $event_price_values, $price_set_amount);
-      $cost = $event_price_values['amount'];
-
       $discountCode = $this->_price_values['discountcode'];
       if (!empty($discountCode)) {
-        $discounted_event_ids = _cividiscount_get_discounted_event_ids();
-        if (in_array($event_in_cart->event_id, $discounted_event_ids)) {
-          $this->apply_discount($discountCode, $price_set_amount, $cost);
+        $ret=$this->apply_discount($discountCode, $price_set_amount, $cost, $event_in_cart->event_id);
+        if ($ret == FALSE) {
+          $cost = $event_price_values['amount'];
         }
+      } else {
```

Comment:
This is one of the things the style warning is complaining about - the standard (from drupal) is

```
}
else {
```

It seems wierd at first but it does help with patching - helps the diff functionality get it right